#!/usr/bin/env python
PACKAGE = "tbf_gripper_perception"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("sm_relSampleStep", 	double_t, 0, "RelativeSamplingStep",	.03, 0,     1)
gen.add("sm_relDistanceStep", 	double_t, 0, "RelativeDistanceStep",	.03, 0,     1)
gen.add("sm_numAngles",    		int_t,    0, "NumberOfAngles",		 	 30, 0,   100)
gen.add("sm_relSceneSampleStep",double_t, 0, "RelativeSceneSampleStep", .20, 0,     1)
gen.add("sm_relSceneDistance", 	double_t, 0, "RelativeSceneDistance",	.03, 0,     1)
gen.add("sm_icp_iterations",    int_t,    0, "Number of ICP iterations",100, 0, 10000)
gen.add("sm_icp_tolerance", 	double_t, 0, "ICP tolerance", 		   .005, 0,     1)
gen.add("sm_icp_rejectionScale",double_t, 0, "ICP rejection scale",  	2.5, 0,   100)
gen.add("sm_icp_numLevels",    	int_t,    0, "Number of ICP levels",  	  8, 0,   100)

exit(gen.generate(PACKAGE, "tbf_gripper_perception", "SurfaceMatching"))